src/init.cpp

```diff
@@ -1365,6 +1351,20 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     } else {
         LogPrintf("Validating signatures for all blocks.\n");
     }
+    arith_uint256 nMinimumChainWork;
+    if (args.IsArgSet("-minimumchainwork")) {
```

Comment:
In commit "Move ::nMinimumChainWork into ChainstateManager" (fa50c08f34ba5ae9f5d9c5df76542f87cb859f54)

Throughout the PR, instead of moving `-checkblockindex/` / `-checkpoints/` / `-assumevalid/` / `-minimumchainwork/` / `-maxtipage` options parsing code from `AppInitParameterInteraction` function to `AppInitMain`, I think it would be better to move it to a function that only parses arguments (`node::ReadValidationArgs(const ArgsManager& args, ChainstateManagerOpts& options)` or similar). As long as the code needs to move anyway, better to move it to a place where it can be indepedently called and tested and doesn't complicate `AppInitMain`.

Reply:
Thx, reworked